src/main/pages/developer-guide/workspace-agent.adoc

```diff
@@ -17,27 +17,61 @@ folder: developer-guide
 
 image::wsagent/wsagent-debug-env.png[]
 
-2) Define a Workspace link:servers.html[Server] for your JPDA port.
+2) Get host and port for connecting to WS Agent Tomcat. This depends on the infrastructure:
 
-image::wsagent/wsagent-debug-server.png[]
+== Connecting to WS Agent tomcat with debugger:
+
+You will need host and port for debugger. The process of it's acquiring differs depends on Che Infrastructure:
 
-3) Get host and port for connecting to WS Agent Tomcat. This depends on the infrastructure:
+#Docker
 
-Docker:
+* Before starting a workspace, define a link:servers.html[Server] for your JPDA port.
+
+image::wsagent/wsagent-debug-server.png[]
 
 * You will need to use port of your host, that is mapped to container's port. It can be found in Workspace Servers of running workspace:
 
 image::wsagent/wsagent-debug-docker.png[]
 
-Local OKD:
+#Kubernetes / Openshift
+
+To connect with debugger from outside of the cluster, use Port Forwarding to forward port from your workspace Pod to the port on your localhost.
+
+Get list of pods and find the name of your workspace pod.
+```
+kubectl get pods --all-namespaces
```

Comment:
won't output of this command contains info about namespace as well? So, we can avoid executing one more command with describing pod